tests/unit/zhmcclient_mock/test_idpool.py

```diff
@@ -26,125 +26,156 @@
-        self._test_free_for_lo_hi(lowest, highest)
-
-    def test_all(self):
-        # Knowing that the chunk size is 10, we focus on the sizes and range
-        # boundaries around that
-        self._test_all_for_lo_hi(0, 0)
-        self._test_all_for_lo_hi(0, 1)
-        self._test_all_for_lo_hi(0, 9)
-        self._test_all_for_lo_hi(0, 10)
-        self._test_all_for_lo_hi(0, 11)
-        self._test_all_for_lo_hi(0, 19)
-        self._test_all_for_lo_hi(0, 20)
-        self._test_all_for_lo_hi(0, 21)
-        self._test_all_for_lo_hi(3, 3)
-        self._test_all_for_lo_hi(3, 4)
-        self._test_all_for_lo_hi(3, 9)
-        self._test_all_for_lo_hi(3, 10)
-        self._test_all_for_lo_hi(3, 11)
-        self._test_all_for_lo_hi(3, 12)
-        self._test_all_for_lo_hi(3, 13)
-        self._test_all_for_lo_hi(3, 14)
-        self._test_all_for_lo_hi(9, 9)
-        self._test_all_for_lo_hi(9, 10)
-        self._test_all_for_lo_hi(9, 11)
-        self._test_all_for_lo_hi(9, 18)
-        self._test_all_for_lo_hi(9, 19)
-        self._test_all_for_lo_hi(9, 20)
-        self._test_all_for_lo_hi(10, 10)
-        self._test_all_for_lo_hi(10, 11)
-        self._test_all_for_lo_hi(10, 19)
-        self._test_all_for_lo_hi(10, 20)
-        self._test_all_for_lo_hi(10, 21)
-        self._test_all_for_lo_hi(11, 11)
-        self._test_all_for_lo_hi(11, 12)
-        self._test_all_for_lo_hi(11, 20)
-        self._test_all_for_lo_hi(11, 21)
-        self._test_all_for_lo_hi(11, 22)
+    pool = IdPool(5, 5)
+
+    with pytest.raises(ValueError):
+        pool.free(4)  # not in range
+
+    with pytest.raises(ValueError):
+        pool.free(5)  # in range but not allocated
+
+    with pytest.raises(ValueError):
+        pool.free(6)  # not in range
+
+
+def test_invalid_free_error_2():
+    """
+    Test invalid IdPool.free_if_allocated() in some variations.
+    """
+
+    pool = IdPool(5, 5)
+
+    pool.free_if_allocated(4)  # not in range (= not allocated)
+
+    pool.free_if_allocated(5)  # in range but not allocated
+
+    pool.free_if_allocated(6)  # not in range (= not allocated)
+
+
+def _test_exhausting_for_lo_hi(lowest, highest):
+    """
+    Internal helper function that tests exhausting an IdPool.
+    """
+
+    start = lowest
+    end = highest + 1
+
+    pool = IdPool(lowest, highest)
+
+    # Exhaust the pool
+    id_list = []
+    for _ in range(start, end):
+        _id = pool.alloc()
+        id_list.append(_id)
+
+    # Verify uniqueness of the ID values
+    id_set = set(id_list)
+    assert len(id_set) == len(id_list)
+
+    # Verify that the pool is exhausted
+    with pytest.raises(ValueError):
+        pool.alloc()
+
+
+def _test_free_for_lo_hi(lowest, highest):
+    """
+    Internal helper function that tests exhausting, freeing, and again
+    exhausting an IdPool.
+    """
+
+    start = lowest
+    end = highest + 1
+
+    pool = IdPool(lowest, highest)
+
+    # Exhaust the pool
+    id_list1 = []
```

Comment:
I suggest to rewrite this as a list comprehension.

Reply:
See previous comment.